packages/build/src/js-transform.ts

```diff
@@ -175,33 +177,41 @@ export function jsTransform(js: string, options: JsTransformOptions): string {
   }
   if (options.transformImportMeta === true ||
       (options.transformImportMeta === undefined &&
-       options.transformModulesToAmd === true)) {
+       options.transformModulesToAmd)) {
     if (!options.filePath) {
       throw new Error('Cannot perform importMeta transform without filePath.');
     }
     if (!options.rootDir) {
       throw new Error('Cannot perform importMeta transform without rootDir.');
     }
-    doBabel = true;
+    doBabelTransform = true;
     let relativeURL = relative(options.rootDir, options.filePath);
     if (isWindows()) {
       // normalize path separators to URL format
       relativeURL = relativeURL.replace(/\\/g, '/');
     }
     plugins.push(rewriteImportMeta(relativeURL));
   }
-  if (options.transformModulesToAmd) {
-    if (options.transformImportMeta === false) {
-      throw new Error(
-          'Cannot use transformModulesToAmd without transformImportMeta.');
-    }
-    doBabel = true;
-    plugins.push(...babelTransformModulesAmd);
+  if (options.transformModulesToAmd && options.transformImportMeta === false) {
+    throw new Error(
+        'Cannot use transformModulesToAmd without transformImportMeta.');
   }
 
-  if (doBabel) {
+  if (doBabelTransform || options.transformModulesToAmd) {
+    let ast;
     try {
-      js = babelCore.transform(js, {presets, plugins}).code!;
+      ast = babylon.parse(js, {
+        // TODO(aomarks) Remove any when typings are updated for babylon 7.
+        sourceType: options.transformModulesToAmd === 'auto' ?
+            'unambiguous' as any :
```

Comment:
Neat! We should do this trick in the analyzer too, to avoid double-parsing module files.